modules/audienceNetworkBidAdapter.js

```diff
@@ -0,0 +1,205 @@
+ * @param {Array} bids - list of bids
+ * @param {String} bids[].placementCode - Prebid placement identifier
+ * @param {Object} bids[].params
+ * @param {String} bids[].params.placementId - Audience Network placement identifier
+ * @param {String} bids[].params.format - Optional format, one of 'video', 'native' or 'fullwidth' if set
+ * @param {Array} bids[].sizes - list of desired advert sizes
+ * @param {Array} bids[].sizes[] - Size arrays [h,w]: should include one of [300, 250], [320, 50]: first matched size is used
+ * @returns {Array<Object>} List of URLs to fetch, plus formats and sizes for later use with interpretResponse
+ */
+const buildRequests = bids => {
+  // Build lists of placementids, adformats, sizes and SDK versions
+  const placementids = [];
+  const adformats = [];
+  const sizes = [];
+  const sdk = [];
+  const requestIds = [];
+
+  bids.forEach(bid => bid.sizes
+    .map(flattenSize)
+    .filter(size => isValidSize(size) || isVideo(bid.params.format))
+    .slice(0, 1)
+    .forEach(size => {
+      placementids.push(bid.params.placementId);
+      adformats.push(bid.params.format || size);
+      sizes.push(size);
+      sdk.push(sdkVersion(bid.params.format));
+      requestIds.push(bid.bidId);
+    })
+  );
+
+  // Build URL
+  const testmode = isTestmode();
+  const pageurl = getTopWindowUrl();
+  const search = {
+    placementids,
+    adformats,
+    testmode,
+    pageurl,
+    sdk
+  };
+  const video = adformats.findIndex(isVideo);
+  if (video !== -1) {
+    [search.playerwidth, search.playerheight] = sizes[video].split('x').map(Number)
+  }
+  const data = formatQS(search);
+
+  return [{ adformats, data, method, requestIds, sizes, url }];
+};
+
+/**
+ * Convert a server response to a bid response.
+ * @param {Object} data - response, already converted from JSON
+ * @param {Object} bidRequests - the original bid requests
+ * @param {Array} bidRequest.adformats - list of formats for the original bid requests
+ * @param {Array} bidRequest.sizes - list of sizes fot the original bid requests
+ * @returns {Array<Object>} A list of bid response objects
+ */
+const interpretResponse = (data, { adformats, requestIds, sizes }) => {
+  const ttl = Number(config.getConfig().bidderTimeout);
+
+  return data.errors && data.errors.length
+    ? []
+    : Object.keys(data.bids)
+      // extract Array of bid responses
+      .map(placementId => data.bids[placementId])
+      // flatten
+      .reduce((a, b) => a.concat(b), [])
+      // transform to bidResponse
+      .map((bid, i) => {
+        const {
+          bid_id: fb_bidid,
+          placement_id: creativeId,
+          bid_price_cents: cpm
+        } = bid;
+
+        const format = adformats[i];
+        const [width, height] = sizes[i];
+        const ad = createAdHtml(creativeId, format, fb_bidid);
+        const requestId = requestIds[i];
+
+        const bidResponse = {
+          // Prebid attributes
+          requestId,
+          bidderCode: code,
```

Comment:
For some subtle reasons, you shouldn't set `bidderCode` in here. It breaks adapter aliasing.

Our docs have been updated to reflect this... but sorry for the confusion all the same.